lib/gooddata/bricks/middleware/logger_middleware.rb

```diff
@@ -4,24 +4,50 @@
 # This source code is licensed under the BSD-style license found in the
 # LICENSE file in the root directory of this source tree.
 
-require 'logger'
-
 require_relative 'base_middleware'
 
 module GoodData
+  class << self
+    attr_accessor :execution_id
+  end
+
   module Bricks
     class LoggerMiddleware < Bricks::Middleware
+      ID_LENGTH = 16
+
+      def generate_string(length = ID_LENGTH)
```

Comment:
Doesn't depend on instance state (maybe move it to another class?) (UtilityFunction)